### PR TITLE
Ecmorris/orbit camera around object

### PR DESCRIPTION
**Pull request by ecmorris**
Orbit Camera around Object sample. QT & QML Implementation

Shows off the functionality of the `OrbitGeoElementCameraController` type. Setting up the scene camera to orbit a plane model, with functionality to transition too/from the cockpit.

https://devtopia.esri.com/runtime/common-samples/issues/1188

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/OrbitCameraAroundObject/OrbitCameraAroundObject.cpp [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+
+  /* Plane model Setup */
+  //Load a plane model.
+  QUrl planeFileURl = QUrl(defaultDataPath() + "/ArcGIS/Runtime/Data/3D/Bristol/Collada/Bristol.dae");
+  ModelSceneSymbol* planeModel = new ModelSceneSymbol(planeFileURl, 1.0, this);
+  planeModel->load();
+
+  //Create a graphic from the plane model, positioned atop a runway.
+  Point runwayPos = Point{6.637, 45.399, 100, SpatialReference::wgs84()};
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/OrbitCameraAroundObject/OrbitCameraAroundObject.cpp]: Our minimum supported Linux compiler is 4.8. As such - do check that initializer braces work as it is temperamental. I've checked for you and can confirm it compiles on Redhat with 4.8.

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/OrbitCameraAroundObject/OrbitCameraAroundObject.qml [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    Rectangle {
+        anchors {
+            bottom: parent.bottom
+            left: parent.left
+            right: parent.right
+            margins: 30
+        }
+
+        height: cameraHeadingSlider.height + cameraHeadingLabel.height
+        color: uiBackgroundCol
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/OrbitCameraAroundObject/OrbitCameraAroundObject.qml]: This sample looks different in terms of style to the other samples so far. Is it possible to bring the styling more in line?

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/OrbitCameraAroundObject/OrbitCameraAroundObject.qml [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        anchors {
+            left: parent.left
+            top: parent.top
+        }
+
+        height: childrenRect.height
+        width: childrenRect.width //Add a little bit of manual padding to comfortably fit the checkbox text
+        color: uiBackgroundCol
+
+        Column {
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/OrbitCameraAroundObject/OrbitCameraAroundObject.qml]: If you use a `ColumnLayout` as opposed to a `Column` you don't need to next an additional `Row`. Just use `Layout.columnSpan: 2` on the buttons.

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/OrbitCameraAroundObject/OrbitCameraAroundObject.cpp (hunk 1 above)
Comment (ecmorris) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/OrbitCameraAroundObject/OrbitCameraAroundObject.cpp]: Ach, that's scary.
D'you recommend defaulting against braced initialization in general?

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/OrbitCameraAroundObject/OrbitCameraAroundObject.cpp [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+    renderer3D->setSceneProperties(properties);
+  }
+  sceneOverlay->setRenderer(renderer3D);
+
+
+  /* Plane model Setup */
+  //Load a plane model.
+  QUrl planeFileURl = QUrl(defaultDataPath() + "/ArcGIS/Runtime/Data/3D/Bristol/Collada/Bristol.dae");
+  ModelSceneSymbol* planeModel = new ModelSceneSymbol(planeFileURl, 1.0, this);
+  planeModel->load();
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/OrbitCameraAroundObject/OrbitCameraAroundObject.cpp]: there shouldn't be a need to explicitly load the model

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/OrbitCameraAroundObject/OrbitCameraAroundObject.cpp [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+
+  //The animation may rotate us over the set camera bounds based on the plane pitch, so unlock them.
+  m_orbitCam->setMinCameraPitchOffset(-180.0);
+  m_orbitCam->setMaxCameraPitchOffset(180.0);
+
+  //Camera move-to-cockpit callback.
+  connect(m_orbitCam, &OrbitGeoElementCameraController::moveCameraCompleted, this,
+          [this](QUuid, bool succeeded)
+  {
+    if(succeeded)
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/OrbitCameraAroundObject/OrbitCameraAroundObject.cpp]: our coding standard calls for a space between the if and the parenthesis - `if (succeeded)`

**Pull request merged**